src/main/java/com/minecolonies/coremod/entity/pathfinding/PathNavigate.java

```diff
@@ -280,42 +280,42 @@ public void onUpdateNavigation()
                             break;
                     }
 
-                    this.entity.getMoveHelper().setMoveTo(vec3.xCoord, vec3.yCoord, vec3.zCoord, newSpeed);
+                    this.ourEntity.getMoveHelper().setMoveTo(vec3.x, vec3.y, vec3.z, newSpeed);
                 }
             }
-            else if (entity.isInWater())
+            else if (ourEntity.isInWater())
             {
                 //  Prevent shortcuts when swimming
                 final int curIndex = this.getPath().getCurrentPathIndex();
                 if (curIndex > 0
                       && (curIndex + 1) < this.getPath().getCurrentPathLength()
-                      && this.getPath().getPathPointFromIndex(curIndex - 1).yCoord != pEx.yCoord)
+                      && this.getPath().getPathPointFromIndex(curIndex - 1).y != pEx.y)
                 {
                     //  Work around the initial 'spin back' when dropping into water
                     oldIndex = curIndex + 1;
                 }
 
                 this.getPath().setCurrentPathIndex(oldIndex);
 
-                Vec3d vec3d = this.getPath().getPosition(this.entity);
+                Vec3d vec3d = this.getPath().getPosition(this.ourEntity);
 
-                if (vec3d.squareDistanceTo(new Vec3d(entity.posX, vec3d.yCoord, entity.posZ)) < 0.1
-                      && Math.abs(entity.posY - vec3d.yCoord) < 0.5)
+                if (vec3d.squareDistanceTo(new Vec3d(ourEntity.posX, vec3d.y, ourEntity.posZ)) < 0.1
+                      && Math.abs(ourEntity.posY - vec3d.y) < 0.5)
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Assign this magic number 0.5 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)